src/MIDebugEngine/Engine.Impl/DebuggedThread.cs

```diff
@@ -151,9 +151,11 @@ internal void ThreadCreatedEvent(int id, string groupId)
         {
             lock (_threadList)
             {
-                var thread = _threadList.Find(t => t.Id == id);
-                if (thread == null)
+                bool bNew;
+                var thread = FindThread(id, out bNew);
+                if (bNew)
                 {
+                    _callback.OnThreadStart(thread);
```

Comment:
The one problem I can see with this is that in CollectThreadsInfo we will not just have the id, but we will also have the 'TargetId'. It looks like the 'TargetId' is what we expose to AD7. Here, we only have the Id. AD7 is not going to be happy if a thread's ID changes.